specs/_features/eip7594/das-core.md

```diff
@@ -239,7 +270,29 @@ To custody a particular column, a node joins the respective gossip subnet. Verif
 
 ## Peer sampling
 
-A node SHOULD maintain a diverse set of peers for each column and each slot by verifying responsiveness to sample queries. At each slot, a node makes `SAMPLES_PER_SLOT` queries for samples from their peers via `DataColumnSidecarsByRoot` request. A node utilizes `get_custody_columns` helper to determine which peer(s) to request from. If a node has enough good/honest peers across all rows and columns, this has a high chance of success.
+### Sample selection
+
+At each slot, a node SHOULD select at least `SAMPLES_PER_SLOT` column IDs for sampling. It is recommended to use uniform random selection without replacement based on local randomness. Sampling is considered successful if the node manages to retrieve all selected columns.
```

Comment:
Selection with or without replace doesn't matter. The security remains indifferent. I think this can mislead the readers that the one without replacement is necessary.

Reply:
I'm quite sure that it actually makes a difference, and thus it is more preferable.
Just take the simple edge-case of NUMBER_OF_COLUMNS=4, SAMPLES_PER_SLOT=2. Without replacement, you would have certainty, since you would have 2 columns from which you can always reconstruct. Allowing replacement there is 1/4 chance you have only 1 column, in which case there is 1/4 chance you do not discover an unavailable block with only 1 column published.

Reply:
I assumed that `NUMBER_OF_COLUMNS` and `SAMPLES_PER_SLOT` are high enough. However, it's fine for me if you want to take care of the low numbers as well.

> Without replacement, you would have certainty, since you would have 2 columns from which you can always reconstruct

Reconstruction shouldn't be considered in the sampling process. In the real setting, `NUMBER_OF_COLUMNS` is 128 and `SAMPLES_PER_SLOT` is 8. `SAMPLES_PER_SLOT` is a lot lower than half of the number of columns, so the reconstruction is impossible for sampling nodes and it's a completely different story from this case.

> Allowing replacement there is 1/4 chance you have only 1 column, in which case there is 1/4 chance you do not discover an unavailable block with only 1 column published.

This is not a fair comparison because it only works in the case of `NUMBER_OF_COLUMNS=4`, if the number is just a bit higher, the one without replacement also has a chance.

![Screenshot from 2024-06-08 16-44-05](https://github.com/ethereum/consensus-specs/assets/631508/58806974-b676-4757-b9db-9a5feba16aba)

Reply:
Low number of columns is definitely not the goal here. The toy example I gave above was specifically just to explain that with or without replacement you get different probabilities. I was trying to highlight that this is just a small example, but maybe it was not clear.

The reason why we recommend selection without replacement is because it is always better than selection with replacement. Depending on parameters, the difference might be small, or bigger. This is not about how big the difference is, just simply good practice to use the better solution, since implementing it is simple.

Note that this is just a recommendation, not a MUST. An implementation might have valid reasons to do it differently.

Reply:
I hope my comments above clarified this. I say "implementing it is simple" because it is mostly a library function, and also example code is widely available. Closing as resolved, but feel free to reopen in case of doubts.